doc/orchestrator.rst

```diff
@@ -165,3 +165,287 @@ A full example of configuring KeyDB to run in SmartSim is shown below
   export REDIS_CONF=/path/to/keydb.conf
 
   # run smartsim workload
+
+Multidb
+=======
+
+To support computationally intensive tasks like scientific simulations,
+SmartSim provides functionality to automate the deployment of multiple
+databases on an HPC cluster. As data volume increases, consider augmenting
+the number of sharded databases or provision multiple standalone databases,
+to ensure the workload can scale to meet resource demands. To ensure that data
+retrieval, updates, and synchronization are carried out smoothly and efficiently
+across multiple database nodes, unique identifier names are given to each database
+during initialization. Below is a general example of launching multiple databases
+with a unique identifier and distributing data between the launched databases.
+
+Defining workflow stages requires the utilization of functions associated
+with the Experiment object. The Experiment object is intended to be instantiated
+once and utilized throughout the workflow runtime. Begin by creating an Experiment
+object and assign it to the 'exp' variable. In this example, we instantiate the
+Experiment object with the name 'getting-started-multidb' and the 'launcher=slurm'
+configuration.
+
+.. code-block:: python
+  import numpy as np
+  from smartredis import Client
+  from smartsim import Experiment
+
+  exp = Experiment("getting-started-multidb", launcher="slurm")
+
+
+In the context of this Experiment, it's essential to create and launch
+the databases as a preliminary step before any other components. We aim
+to showcase the multi-database automation capabilities of SmartSim, so we
+create two types of databases in the workflow: a standalone database and a
+clustered database.
+
+To create a database, you can utilize the create_database() function located
+in the Experiment class. For launching multiple databases, this function requires
+specifying a unique database identifier argument named db_identifier. The db_identifier
+argument plays a crucial role in SmartSim multi-database support by serving as a unique
+identifier for each database being launched. The db_identifier will be utilized later
+in the example.
+
+Here's how to create the databases:
+
+For the standalone database:
+
+.. code-block:: python
+  standalone_database = exp.create_database(port=6379, db_nodes=1, interface="ib0", db_identifier="std-deployment")
+  exp.generate(standalone_database, overwrite=True)
+
+For the clustered database:
+
+.. code-block:: python
+  clustered_database = exp.create_database(port=6380, db_nodes=3, interface="ib0", db_identifier="clus-deployment")
```

Comment:
hyphens in the db_identifier implies hyphens in the environment variable names. I _think_ it's supported but it makes me anxious

Reply:
I would say yes. You could also mention the specific environment variables that get created

Reply:
yes perfect!